packages/item-cc/src/item.controller.ts

```diff
@@ -155,15 +164,67 @@ export class ItemController extends ConvectorController {
   ) {
     //first check if item exists
     let item = await Item.getOne(itemId);
+
+    if (item.proposalAccepted) {
+      throw new Error('This proposal was already accepted, this cannot be changed anymore.');
+    }
+
     await ItemController.checkValidItem(item);
     await ItemController.checkValidOwner(this.sender, item.proposedOwner);
 
     //If the proposal is accepted, change the item owner
     if (accept) {
-      item.itemHistory.push(new TransferEvent(item.proposedOwner, item.name, item.itemOwner));
-      item.itemOwner = item.proposedOwner;
+      item.proposalAccepted = true;
+    } else {
+      item.proposalAccepted = false;
+      item.proposedOwner = '';
```

Comment:
This line should always be executed, whether it gets accepted or not

Reply:
No because the item goes to the transporter first, so you don't change ownership yet, therefore you still need the proposedOwner otherwise the transporter does not know where to send the item